### DISCUSSION TITLE
What application is used to generate the state machine demo in the docs?

### DISCUSSION THREAD

**Discussion by DylanSp** [microsoft/coyote#396] [Q&A] [answered]
Is the source code for the application/test used for the [state machine demo](https://microsoft.github.io/coyote/#concepts/actors/state-machine-demo/) available? I'm curious how the invariant that there should be at most one Leader at a time is communicated to the test.

**Comment by lovettchris** [comment, marked as answer]
Yes, the sample code is [available here](https://github.com/microsoft/coyote/tree/main/Samples/CloudMessaging).  The code has evolved a bit since the diagram was generated, but you can see the [SafetyMonitor ](https://github.com/microsoft/coyote/blob/main/Samples/CloudMessaging/Raft.Mocking/Specifications/SafetyMonitor.cs) is the one that is ensuring there is only one leader per "Term", where Term is a concept of version or time in the Raft protocol.  This is for the "Mock" version where testing of the Raft protocol is done all in one process.

**Comment by DylanSp** [comment]
Got it, thanks!
